app/controllers/user/comments_controller.rb

```diff
@@ -1,11 +1,27 @@
 class User::CommentsController < ApplicationController
+  rescue_from ActiveRecord::RecordNotFound, with: :respond_bad_request
+
   def index
-    posts = Post.all
-    comments = posts.map(&:comments).flatten
-    @user_comments = comments.select do |comment|
-      comment.author.username == params[:username]
-    end
+    author = User.find_by!(username: params.require(:username))
+    comments = Comment.where(author: author)
+                      .limit(pagination_params[:limit])
+                      .offset(pagination_params[:offset])
+    render json: comments
+  end
+
+  private
+
+  def respond_bad_request
```

Comment:
This updates the original behavior to return an error message if the user doesn't exist. The error message should be generic to avoid creating security issues.